app/models/package_manager/maven.rb

```diff
@@ -230,20 +232,20 @@ def self.licenses(xml)
         .map(&:last)
     end
 
-    def self.latest_version(versions, name)
-      if versions.present?
-        versions
-          .max_by { |version| version[:published_at] }
-          .dig(:number)
-      else
-        # TODO: this is in place to handle packages that are no longer on maven-repository.com
-        # this could be removed if we switched to a package data provider that supplied full information
-        Project
-          .find_by(name: name, platform: formatted_name)
-          &.versions
-          &.max_by(&:published_at)
-          &.number
-      end
+    def self.latest_version(name)
+      maven_metadata = get_raw(MavenUrl.from_name(name, repository_base, NAME_DELIMITER).maven_metadata)
```

Comment:
This fetches the maven metadata separately from `version()` (so we'll call it twice -- should be cheap, but is a good use-case for maintaining state in PackageManager::Base, where we could cache the maven metadata in the instance).